### PR TITLE
[Locked Figure Labels] Add labels to locked lines' defining points in the graph and editor

### PR DESCRIPTION
**Pull request by nishasy**
## Summary:
Add support for locked labels settings within the locked
points that define locked lines.

This allows content authors to directly add labels to
the points on either end of a line/ray/segment instead
of having to create two labels on the line and manually
position them. This also means the data contains more
information about how the labels are being used within
the context of the line.

I also deleted `DefiningPointSettings` so that I wouldn't
have to reimplement all the logic for adding/updating/removing
the locked labels associated with a locked point. Instead, I
consolidated it all into `LockedPointSettings`.

Issue: https://khanacademy.atlassian.net/browse/LEMS-2347

## Test plan:
- Go to http://localhost:6006/?path=/story/perseuseditor-widgets-interactive-graph--mafs-with-locked-labels-flag
- Confirm that neither locked points nor locked lines have visible labels in the editor or the graph
- Go to http://localhost:6006/?path=/story/perseuseditor-widgets-interactive-graph--mafs-with-locked-point-labels-flag
- Confirm that only locked points have visible labels in the editor and the graph
- Play around with the labels and confirm that they look right and update accordingly with the point settings
- Go to http://localhost:6006/?path=/story/perseuseditor-widgets-interactive-graph--mafs-with-locked-line-labels-flag
- Confirm that only the locked lines have visible labels in the editor and graph
- Play around with the labels and confirm that they look right and update accordingly with the line settings

## Screenshots:
| Labels | Points | Lines |
| --- | --- | --- |
| Screenshot 2024-09-18 at 3 28 36 PM | Screenshot 2024-09-18 at 3 28 49 PM | Screenshot 2024-09-18 at 3 29 22 PM |

image

### CODE REVIEW

File: packages/perseus-editor/src/widgets/interactive-graph-editor/locked-figures/locked-label-settings.tsx [hunk 1]
```diff
@@ -192,7 +197,4 @@ const styles = StyleSheet.create({
     spaceUnder: {
         marginBottom: spacing.xSmall_8,
     },
-    innerLabelAccordionPanel: {
```
Comment (catandthemachines) [inline packages/perseus-editor/src/widgets/interactive-graph-editor/locked-figures/locked-label-settings.tsx:195]: Question: why remove this styling?

File: packages/perseus-editor/src/widgets/interactive-graph-editor/locked-figures/locked-point-settings.tsx [hunk 2]
```diff
@@ -23,29 +23,79 @@ import LockedFigureSettingsActions from "./locked-figure-settings-actions";
...
 
-export type Props = LockedFigureSettingsCommonProps &
-    LockedPointType & {
-        /**
-         * Called when the props (coords, color, etc.) are updated.
-         */
-        onChangeProps: (newProps: Partial<LockedPointType>) => void;
-    };
+export type Props = LockedPointType & {
+    flags?: APIOptions["flags"];
```
Comment (catandthemachines) [inline packages/perseus-editor/src/widgets/interactive-graph-editor/locked-figures/locked-point-settings.tsx:34]: Comment please ❤️

**Comment by catandthemachines** [comment]
I noticed while playing around, that the example where only the line has a label does not allow me to add a label to an existing point. Is this by design?
image
https://650db21c3f5d1b2f13c02952-jslixzchku.chromatic.com/?path=/story/perseuseditor-widgets-interactive-graph--mafs-with-locked-line-labels-flag

**Review by catandthemachines — APPROVED** [review]
Overall the code looks clean! Approving with some additional questions on the experience and some nits to clean up ❤️

**Comment by nishasy** [comment]
@catandthemachines 
> I noticed while playing around, that the example where only the line has a label does not allow me to add a label to an existing point. Is this by design?

Yup! The "locked line labels flag" story _only_ has locked lines' labels enabled, not locked points' labels. I did it this way in case we end up switching on one flag but not the other. I was using this Storybook structure to manually test that the flags are separated correctly.

**Pull request merged**